runtime/doc/lua.txt

```diff
@@ -698,25 +698,26 @@ VIM.HIGHLIGHT                  				*lua-highlight*
 Nvim includes a function for highlighting a selection on yank (see for example
 https://github.com/machakann/vim-highlightedyank). To enable it, add
 >
- au TextYankPost * silent! lua require'vim.highlight'.on_yank()
+ au TextYankPost * silent! vim.highlight.on_yank()
 <
 to your `init.vim`. You can customize the highlight group and the duration of
 the highlight via
 >
- au TextYankPost * silent! lua require'vim.highlight'.on_yank("IncSearch", 500)
+ au TextYankPost * silent! vim.highlight.on_yank {higroup="IncSearch", timeout=250}
 <
 If you want to exclude visual selections from highlighting on yank, use
 >
-au TextYankPost * silent! lua return (not vim.v.event.visual) and require'vim.highlight'.on_yank()
+ au TextYankPost * silent! vim.highlight.on_yank {on_visual=false}
 <
 
-vim.highlight.on_yank([{higroup}, {timeout}, {event}]) 
-                                                    *vim.highlight.on_yank()*
-        Highlights the yanked text. Optional arguments are the highlight group
-        to use ({higroup}, default `"IncSearch"`), the duration of highlighting
-        in milliseconds ({timeout}, default `500`), and the event structure 
-        that is fired ({event}, default `vim.v.event`).
-
+vim.highlight.on_yank({opts})                         *vim.highlight.on_yank()*
+        Highlights the yanked text. The fields of the optional dict {opts}
+        control the highlight: the highlight group to use ({higroup}, default
+        `"IncSearch"`), the duration of highlighting in milliseconds
+        ({timeout}, default `250`), whether to highlight a visual selection
+        ({on_visual}, default `true`), whether to highlight when executing a
+        macro ({on_macro}, default `false`), and the {event} (struct, default
+        `vim.v.event`).
```

Comment:
I think this section should just be listed out like how you have it in the doc string.

It's easier to grok each of the keys that way IMO